### DISCUSSION TITLE
Authorized-only access to S3 data in self hosted Outline

### DISCUSSION THREAD

**Discussion by vaishnavsm** [outline/outline#2677] [Self Hosting] [answered]
Hey guys!

I'm self hosting an outline instance, and I wanted to know if there was any way to prevent unauthorized access to the S3 data.
I've set the CORS rule to allow `GET` from `*`, but this means that people who aren't logged in to outline can see the data by using the url of the resource (image, etc). 

I think the standard way this is solved is to proxy the request through the attached service account so that only the Outline server actually hits the s3 endpoint?

Maybe there are alternative solutions that we can use?

Thanks!

**Comment by tommoor** [comment, marked as answer]
The default is to upload attachments with private ACL, Outline generates signed urls for access. Have you set `AWS_S3_ACL` to something other than `private`?

The only exception to this is avatar/profile images which are uploaded as public.

**Reply by vaishnavsm** [reply]
`AWS_S3_ACL` is `private`, but the urls of the embeddings are using the s3 url directly. This may have to do with the script I'd written to upload existing images to outline, perhaps.

**Reply by vaishnavsm** [reply]
Let me verify this

**Reply by vaishnavsm** [reply]
You're right. When I add new attachments, they seem to be proxied. Now I'll just go spend some time writing a script which will reupload everything for me automatically using the API. Will share here if anyone needs it to migrate resources!

**Reply by vaishnavsm** [reply]
Thanks! Wonderful work here :)
